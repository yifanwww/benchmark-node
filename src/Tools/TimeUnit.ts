export enum TimeUnit {
    NS,
    US,
    MS,
    S,
}

const TimeUnitStr = {
    [TimeUnit.NS]: 'ns',
    [TimeUnit.US]: 'us',
    [TimeUnit.MS]: 'ms',
    [TimeUnit.S]: 's',
};

const TimeUnitDescription = {
    [TimeUnit.NS]: '1 Nanosecond (0.000000001 sec)',
    [TimeUnit.US]: '1 Microsecond (0.000001 sec)',
    [TimeUnit.MS]: '1 Millisecond (0.001 sec)',
    [TimeUnit.S]: '1 Second',
};

export class TimeUnitHelper {
    static getUnitStr = (unit: TimeUnit) => TimeUnitStr[unit];

    static getUnitDescription = (unit: TimeUnit) => TimeUnitDescription[unit];

    static getFullDescription = (unit: TimeUnit, maxHeaderLen: number) =>
        `1 ${TimeUnitHelper.getUnitStr(unit).padEnd(maxHeaderLen - 2)}: ${TimeUnitHelper.getUnitDescription(unit)}`;

    static chooseUnit(time: number): TimeUnit {
        if (time <= 1e3) {
            return TimeUnit.NS;
        } else if (time <= 1e6) {
            return TimeUnit.US;
        } else if (time <= 1e9) {
            return TimeUnit.MS;
        } else {
            return TimeUnit.S;
        }
    }
}
