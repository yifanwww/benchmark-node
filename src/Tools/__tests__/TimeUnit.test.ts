import { createClassJestHelper } from '../../__tests__/class';
import { TimeUnit, TimeUnitHelper } from '../TimeUnit';

const { buildStaticMethodName } = createClassJestHelper(TimeUnitHelper);

describe(`Test static method \`${buildStaticMethodName('getUnitStr')}\``, () => {
    it('should get time unit string', () => {
        expect(TimeUnitHelper.getUnitStr(TimeUnit.NS)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitStr(TimeUnit.US)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitStr(TimeUnit.MS)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitStr(TimeUnit.S)).toMatchSnapshot();
    });
});

describe(`Test static method \`${buildStaticMethodName('getUnitDescription')}\``, () => {
    it('should get time unit description', () => {
        expect(TimeUnitHelper.getUnitDescription(TimeUnit.NS)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitDescription(TimeUnit.US)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitDescription(TimeUnit.MS)).toMatchSnapshot();
        expect(TimeUnitHelper.getUnitDescription(TimeUnit.S)).toMatchSnapshot();
    });
});

describe(`Test static method \`${buildStaticMethodName('getFullDescription')}\``, () => {
    it('should get time unit full description', () => {
        expect(TimeUnitHelper.getFullDescription(TimeUnit.NS, 6)).toMatchSnapshot();
        expect(TimeUnitHelper.getFullDescription(TimeUnit.US, 6)).toMatchSnapshot();
        expect(TimeUnitHelper.getFullDescription(TimeUnit.MS, 6)).toMatchSnapshot();
        expect(TimeUnitHelper.getFullDescription(TimeUnit.S, 6)).toMatchSnapshot();
    });
});

describe(`Test static method \`${buildStaticMethodName('chooseUnit')}\``, () => {
    it('should choose a suitable time unit', () => {
        expect(TimeUnitHelper.chooseUnit(0)).toBe(TimeUnit.NS);
        expect(TimeUnitHelper.chooseUnit(1e3)).toBe(TimeUnit.NS);
        expect(TimeUnitHelper.chooseUnit(1e3 + 1)).toBe(TimeUnit.US);
        expect(TimeUnitHelper.chooseUnit(1e6)).toBe(TimeUnit.US);
        expect(TimeUnitHelper.chooseUnit(1e6 + 1)).toBe(TimeUnit.MS);
        expect(TimeUnitHelper.chooseUnit(1e9)).toBe(TimeUnit.MS);
        expect(TimeUnitHelper.chooseUnit(1e9 + 1)).toBe(TimeUnit.S);
    });
});
