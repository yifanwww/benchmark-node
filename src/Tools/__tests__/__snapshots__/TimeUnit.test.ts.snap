// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test class \`TimeUnitHelper\` calls static method getFullDescription 1`] = `"1 ns  : 1 Nanosecond (0.000000001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getFullDescription 2`] = `"1 us  : 1 Microsecond (0.000001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getFullDescription 3`] = `"1 ms  : 1 Millisecond (0.001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getFullDescription 4`] = `"1 s   : 1 Second"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitDescription 1`] = `"1 Nanosecond (0.000000001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitDescription 2`] = `"1 Microsecond (0.000001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitDescription 3`] = `"1 Millisecond (0.001 sec)"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitDescription 4`] = `"1 Second"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitStr 1`] = `"ns"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitStr 2`] = `"us"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitStr 3`] = `"ms"`;

exports[`Test class \`TimeUnitHelper\` calls static method getUnitStr 4`] = `"s"`;
